Add HomelessSwap multi-chain TVL adapter

Hamza-Alarja opened the pull request:
##### Name :  
HomelesSwap

##### Twitter Link:  
https://x.com/homelesswap?lang=en

##### List of audit links if any:  
https://drive.google.com/file/d/1CMmBkVVLX72DDJkogTf7esC80yrvbBqn/view

##### Website Link:  
https://www.homelesswap.finance/

##### Logo (High resolution, will be shown with rounded borders):  
(https://drive.google.com/file/d/1Cyb3najyn6eMFZ6ZnFS3HCTUbtiwdkzN/view?usp=drive_link)

##### Current TVL:  
N/A

##### Treasury Addresses (if the protocol has treasury):  
0x982E680B25c1e0e02b0b84d8f1E008094F200cd5

##### Chain:  
Ethereum, BSC, zkSync, Polygon zkEVM

##### Coingecko ID:  
N/A

##### Coinmarketcap ID:  
N/A

##### Short Description (to be shown on DefiLlama):  
HomelessSwap is a decentralized multi-chain swap protocol designed to offer fast, low-fee trading experiences across Ethereum, BNB Chain, zkSync, and Polygon zkEVM.

##### Token address and ticker if any:  
0x299F467665e1870A705099AA5a0F11520df026bC (HLS)

##### Category:  
DEX

##### Oracle Provider(s):  
N/A

##### Implementation Details:  
N/A

##### Documentation/Proof:  
N/A

##### forkedFrom (pancakeswap):  
N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):  
TVL is calculated by summing the balances of stablecoins (USDC, USDT, BUSD) and wrapped tokens (WETH, WBNB) held within swap commission smart contracts on each supported chain.

##### Github org/user (Optional, if your code is open source):  
https://github.com/Hamza-Alarja/homeless

g1nt0ki commented on projects/homelesswap/index.js:
```diff
@@ -0,0 +1,48 @@
+
+  polygon_zkevm: {
+    tvl: sumTokensExport({
+      owner: '0x0Ad3DaE92e136D363864CCd10c794B36922ccEa0',
```
is this address correct?

g1nt0ki commented on projects/homelesswap/index.js:
```diff
@@ -0,0 +1,48 @@
+const { sumTokensExport } = require('../helper/sumTokens')
+
+module.exports = {
+  zksync: {
```
should be `era`

g1nt0ki commented on projects/homelessswap/index.js:
why is there a duplicate adapter file

g1nt0ki reviewed (changes requested):
thanks for the pr, does the project have any tvl?

llamabutler commented:
Error while running adapter at projects/homelesswap: 
        
 
  
    
    
    SyntaxError: Unexpected end of input
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:122:12
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:239:3)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

Hamza-Alarja commented:
Thanks for the feedback. I’ve removed the token causing issues on Polygon zkEVM for now. Will revisit and add more tokens once confirmed.

Hamza-Alarja commented:
Thanks for the feedback. I’ve removed the token causing issues on Polygon
zkEVM for now. Will revisit and add more tokens once confirmed.

On Tue, Jul 15, 2025 at 6:14 PM waynebruce0x ***@***.***>
wrote:

> *waynebruce0x* left a comment (DefiLlama/DefiLlama-Adapters#15466)
> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/15466#issuecomment-3073784912>
>
>  [chain: polygon_zkevm] [fail count: 2] [abi: function balanceOf(address account) view returns (uint256)]
>     target: 0xb24a0ed5084817de0cd20e2f07b0196ef5daa1e3 params: 0x0ad3dae92e136d363864ccd10c794b36922ccea0]
>     target: 0x4200000000000000000000000000000000000006 params: 0x0ad3dae92e1...] {
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/15466#issuecomment-3073784912>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZDTTSAJHO6MIBH4KPTB7UL3IUED3AVCNFSM6AAAAACBMYMWHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANZTG44DIOJRGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Hamza-Alarja commented:
fixed

On Thu, Jul 17, 2025 at 4:22 PM Mc Larwnce ***@***.***> wrote:

> Thanks for the feedback. I’ve removed the token causing issues on Polygon
> zkEVM for now. Will revisit and add more tokens once confirmed.
>
> On Tue, Jul 15, 2025 at 6:14 PM waynebruce0x ***@***.***>
> wrote:
>
>> *waynebruce0x* left a comment (DefiLlama/DefiLlama-Adapters#15466)
>> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/15466#issuecomment-3073784912>
>>
>>  [chain: polygon_zkevm] [fail count: 2] [abi: function balanceOf(address account) view returns (uint256)]
>>     target: 0xb24a0ed5084817de0cd20e2f07b0196ef5daa1e3 params: 0x0ad3dae92e136d363864ccd10c794b36922ccea0]
>>     target: 0x4200000000000000000000000000000000000006 params: 0x0ad3dae92e1...] {
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/DefiLlama/DefiLlama-Adapters/pull/15466#issuecomment-3073784912>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AZDTTSAJHO6MIBH4KPTB7UL3IUED3AVCNFSM6AAAAACBMYMWHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANZTG44DIOJRGI>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

waynebruce0x commented:
We need non zero TVL to list

Hamza-Alarja commented:
Currently, the protocol is still growing and we’re working on increasing
liquidity and TVL across supported chains. Meanwhile, we can provide any
data or proof needed to verify activity on the platform.

We’ll update the adapter once the TVL is consistently non-zero. Appreciate
your patience and support
[…](https://github.com/DefiLlama/DefiLlama-Adapters/pull/15466#)

g1nt0ki commented:
closing this for now, can re-open when there is some tvl
